src/xradio/vis/_vis_utils/_ms/_tables/create_field_and_source_xds.py

```diff
@@ -464,65 +439,96 @@ def extract_source_info(xds, path, is_ephemeris, source_id, spectral_window_id):
         The xarray dataset with the added source information.
     """
 
-    if source_id == -1:
+    is_ephemeris = xds.attrs[
+        "is_ephemeris"
+    ]  # If ephemeris data is present we ignore the SOURCE_DIRECTION in the source table.
+
+    if all(source_id == -1):
         logger.warning(
             f"Source_id is -1. No source information will be included in the field_and_source_xds."
         )
-        xds.attrs["source_name"] = "None"
+        xds = xds.assign_coords(
+            {"source_name": "Unknown"}
+        )  # Need to add this for ps.summary() to work.
         return xds
 
+    from xradio._utils.list_and_array import check_if_consistent, unique_1d
+
+    unique_source_id = unique_1d(source_id)
+    taql_where = f"where (SOURCE_ID IN [{','.join(map(str, unique_source_id))}]) AND (SPECTRAL_WINDOW_ID = {spectral_window_id})"
+
     source_xds = read_generic_table(
         path,
         "SOURCE",
         ignore=["SOURCE_MODEL"],  # Trying to read SOURCE_MODEL causes an error.
-        taql_where=f"where SOURCE_ID = {source_id} AND SPECTRAL_WINDOW_ID = {spectral_window_id}",
+        taql_where=taql_where,
     )
 
     if len(source_xds.data_vars) == 0:  # The source xds is empty.
         logger.warning(
             f"SOURCE table empty for source_id {source_id} and spectral_window_id {spectral_window_id}."
```

Comment:
I think that as a result of the changes done here, this warning message is now printing very long arrays in `{source_id}` (with many repetitions of the IDs). I think that the intention is to print just the unique IDs, if that's the case I can try to adjust that when fixing other related issues.

Reply:
Thanks, that works.